osu.Game/Graphics/Containers/OsuScrollContainer.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Input;
+using OpenTK.Input;
+
+namespace osu.Game.Graphics.Containers
+{
+    internal class OsuScrollContainer : ScrollContainer
+    {
+        /// <summary>
+        /// Add the ability to seek to an absolute scroll position when the right mouse button is pressed or dragged.
+        /// Uses the value of <see cref="DistanceDecayOnRightMouseScrollbar"/> to smoothly scroll to the dragged location.
+        /// </summary>
+        public bool RightMouseScrollbar = false;

```

Comment:
In that case you should also rename `shouldPerformRelativeDrag`. And what's with the `Bar` added to the end of this? Why not `RightMouseScroll`?